lightning-graph-sync/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+//!
+//! If a particular channel update had never occurred before, the full update is sent. If a channel
+//! has had updates prior to the provided timestamp, the latest update prior to the timestamp is
+//! taken as a reference, and the delta is calculated against it.
+//!
+//! In practice, snapshots of server data will be calculated at regular intervals, such that deltas
+//! are cached and needn't be calculated on the fly.
+
+// Allow and import test features for benching
+#![cfg_attr(all(test, feature = "_bench_unstable"), feature(test))]
+#[cfg(all(test, feature = "_bench_unstable"))] extern crate test;
+
+use std::fs::File;
+
+use lightning::routing::network_graph;
+
+use crate::error::GraphSyncError;
+
+/// Error types that these functions can return
+pub mod error;
+
+/// Core functionality of this crate
+pub mod processing;
+
+/// Sync gossip data from a file
+///
+/// `network_graph`: The network graph to apply the updates to
+///
+/// `sync_path`: Path to the file where the gossip update data is located
+///
+pub fn sync_network_graph_with_file_path(
+	network_graph: &network_graph::NetworkGraph,
+	sync_path: &str,
+) -> Result<(), GraphSyncError> {
+	let mut file = File::open(sync_path)?;
+	processing::update_network_graph_from_byte_stream(&network_graph, &mut file)
+}
+
+#[cfg(test)]
+mod tests {
+	use std::fs;
+
+	use bitcoin::blockdata::constants::genesis_block;
+	use bitcoin::Network;
+
+	use lightning::routing::network_graph::NetworkGraph;
+
+	use crate::sync_network_graph_with_file_path;
+
+	#[test]
+	fn test_sync_from_file() {
+		// same as incremental_only_update_fails_without_prior_same_direction_updates
+		let valid_response = vec![76, 68, 75, 1, 111, 226, 140, 10, 182, 241, 179, 114, 193, 166, 162, 70, 174, 99, 247, 79, 147, 30, 131, 101, 225, 90, 8, 156, 104, 214, 25, 0, 0, 0, 0, 0, 97, 227, 98, 218, 0, 0, 0, 4, 2, 22, 7, 207, 206, 25, 164, 197, 231, 230, 231, 56, 102, 61, 250, 251, 187, 172, 38, 46, 79, 247, 108, 44, 155, 48, 219, 238, 252, 53, 192, 6, 67, 2, 36, 125, 157, 176, 223, 175, 234, 116, 94, 248, 201, 225, 97, 235, 50, 47, 115, 172, 63, 136, 88, 216, 115, 11, 111, 217, 114, 84, 116, 124, 231, 107, 2, 158, 1, 242, 121, 152, 106, 204, 131, 186, 35, 93, 70, 216, 10, 237, 224, 183, 89, 95, 65, 3, 83, 185, 58, 138, 181, 64, 187, 103, 127, 68, 50, 2, 201, 19, 17, 138, 136, 149, 185, 226, 156, 137, 175, 110, 32, 237, 0, 217, 90, 31, 100, 228, 149, 46, 219, 175, 168, 77, 4, 143, 38, 128, 76, 97, 0, 0, 0, 2, 0, 0, 255, 8, 153, 192, 0, 2, 27, 0, 0, 0, 1, 0, 0, 255, 2, 68, 226, 0, 6, 11, 0, 1, 2, 3, 0, 0, 0, 2, 0, 40, 0, 0, 0, 0, 0, 0, 3, 232, 0, 0, 3, 232, 0, 0, 0, 1, 0, 0, 0, 0, 58, 85, 116, 216, 255, 8, 153, 192, 0, 2, 27, 0, 0, 25, 0, 0, 0, 1, 0, 0, 0, 125, 255, 2, 68, 226, 0, 6, 11, 0, 1, 5, 0, 0, 0, 0, 29, 129, 25, 192];
+
+		fs::create_dir_all("./tmp/graph-sync-tests").unwrap();
+		fs::write("./tmp/graph-sync-tests/test_data.lngossip", valid_response).unwrap();
```

Comment:
This leaves a file (and directory) in my repo. Could you clean up and handle failed tests cases similar to what we do in the background processor tests?

https://github.com/lightningdevkit/rust-lightning/blob/d166044f2723ca8602ce654f2c12d4921d0ced9b/lightning-background-processor/src/lib.rs#L419-L427